src/main/java/mezz/jei/ingredients/IngredientListElementComparator.java

```diff
@@ -1,41 +1,483 @@
 package mezz.jei.ingredients;
 
-import java.util.Comparator;
-
 import mezz.jei.config.Constants;
 import mezz.jei.gui.ingredients.IIngredientListElement;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemTool;
+import mezz.jei.config.Config;
+import mezz.jei.util.Translator;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.ai.attributes.AttributeModifier;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemArmor;
+import net.minecraft.item.ItemHoe;
+import net.minecraft.item.ItemShears;
+import scala.Int;
+import java.util.*;
+
+import com.google.common.collect.Multimap;
 
 public final class IngredientListElementComparator implements Comparator<IIngredientListElement> {
 	public static final IngredientListElementComparator INSTANCE = new IngredientListElementComparator();
 
+	public static ArrayList<SortEntry> entries = new ArrayList<SortEntry>();
+	public static ArrayList<SortEntry> list = new ArrayList<SortEntry>();
+	private static boolean areDefaultEntriesLoaded = false;
+
 	private IngredientListElementComparator() {
 
 	}
 
+	public static class SortEntry {
+		public String name;
+		// External interfaces only have access to ItemStacks, sorry fluids.
```

Comment:
I think we should be able to sort any type of ingredient

Reply:
Is IIngredientListElement is something accessible to modders with the API?
If so, there is no reason that the API can't offer that as well.

Reply:
The API can only access classes in the `mezz.jei.api` package, they can't access `IIngredientListElement`.

I think you could make the sorter take an `Object` instead of `ItemStack`, then they can check if it's an ingredient type they understand and sort it, or just return 0 if they don't want to deal with that type of ingredient.

Reply:
What would I be sending the comparator?
After a little bit of review, I think I would be sending the IIngredientListElement.getIngredient() values.  I'm going to have to think.